Cannot build rand 0.6 in a no-std environment.

Hi!

It does not seem to be possible to compile rand 0.6 in a no-std environment.
I discovered this while trying to compile the nalgebra crate with `#[no-std]` and after upgrading from rand 0.5 to 0.6.

Here are the steps I followed to reproduce this independently from nalgebra:

```sh
rustup component add rust-src
cargo install xargo
xargo build --verbose --no-default-features --target=x86_64-unknown-linux-gnu
```

Error message:
```
    Updating crates.io index
   Compiling autocfg v0.1.2
   Compiling libc v0.2.48
   Compiling rand_core v0.4.0 (/Users/sebcrozet/work/rand/rand_core)
error[E0463]: can't find crate for `std`
error[E0463]: can't find crate for `std`


error: aborting due to previous error

error: aborting due to previous error

For more information about this error, try `rustc --explain E0463`.
For more information about this error, try `rustc --explain E0463`.
error: Could not compile `autocfg`.
warning: build failed, waiting for other jobs to finish...
error: Could not compile `libc`.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

Note that this worked fine with rand 0.5.

Thanks for reporting @sebcrozet. I can reproduce, though I get a different error:
```
error[E0463]: can't find crate for `std`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0463`.
error: Could not compile `libc`.

Caused by:
  process didn't exit successfully: `rustc --crate-name libc /home/dhardy/.cargo/registry/src/github.com-1ecc6299db9ec823/libc-0.2.48/src/lib.rs --color always --crate-type lib --emit=dep-info,link -C debuginfo=2 --cfg 'feature="default"' --cfg 'feature="use_std"' -C metadata=47a8dfebb704abf5 -C extra-filename=-47a8dfebb704abf5 --out-dir /home/dhardy/projects/other/nalgebra/target/x86_64-unknown-linux-gnu/debug/deps --target x86_64-unknown-linux-gnu -L dependency=/home/dhardy/projects/other/nalgebra/target/x86_64-unknown-linux-gnu/debug/deps -L dependency=/home/dhardy/projects/other/nalgebra/target/debug/deps --cap-lints allow --sysroot /home/dhardy/.xargo/HOST --cfg core_cvoid` (exit code: 1)
```
(Multiple crates get built in parallel, so the one mentioned just before the error isn't necessarily responsible.)

So it seems that `libc` is being built with default features, which implies `std`. Why is this? This is the tricky bit: `Cargo.lock` lists all dependencies, but not which are enabled or which features are used.

One thing I do notice is that `rust-crypto` depends on `rand` version `0.3.23`. This could be your problem, because before `0.4`, Rand did not support `no_std`.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -9,6 +9,10 @@ A [separate changelog is kept for rand_core](rand_core/CHANGELOG.md).
 You may also find the [Upgrade Guide](https://rust-random.github.io/book/update.html) useful.
 
 
+## Unreleased
+- Enable fork protection of ReseedingRng without `std`
+- Remove dependency on `winapi`
+
 ## [0.6.5] - 2019-01-28
 ### Crates
 - Update `rand_core` to 0.4 (#703)

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -65,11 +65,8 @@ optional = true
 features = ["into_bits"]
 
 [target.'cfg(unix)'.dependencies]
+# Used for fork protection (reseeding.rs)
 libc = { version = "0.2", default-features = false }
-
-# TODO: check if all features are required
-[target.'cfg(windows)'.dependencies]
-winapi = { version = "0.3", features = ["minwindef", "ntsecapi", "profileapi", "winnt"] }
 
 [dev-dependencies]
 # This has a histogram implementation used for testing uniformity.

--- a/rand_jitter/CHANGELOG.md
+++ b/rand_jitter/CHANGELOG.md
@@ -4,6 +4,9 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [0.1.3] - 2019-02-05
+- Use libc in `no_std` mode to fix #723
+
 ## [0.1.2] - 2019-01-31
 - Fix for older rustc compilers on Windows (#722)
 

--- a/rand_jitter/Cargo.toml
+++ b/rand_jitter/Cargo.toml
@@ -18,7 +18,9 @@ rand_core = { path = "../rand_core", version = "0.4" }
 log = { version = "0.4", optional = true }
 
 [target.'cfg(any(target_os = "macos", target_os = "ios"))'.dependencies]
-libc = "0.2"
+# We don't need the 'use_std' feature and depending on it causes
+# issues due to: https://github.com/rust-lang/cargo/issues/1197
+libc = { version = "0.2", default_features = false }
 
 [target.'cfg(target_os = "windows")'.dependencies]
 winapi = { version = "0.3", features = ["profileapi"] }

--- a/rand_jitter/Cargo.toml
+++ b/rand_jitter/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "rand_jitter"
-version = "0.1.2"
+version = "0.1.3"
 authors = ["The Rand Project Developers"]
 license = "MIT/Apache-2.0"
 readme = "README.md"

--- a/src/rngs/adapter/reseeding.rs
+++ b/src/rngs/adapter/reseeding.rs
@@ -282,12 +282,12 @@ where R: BlockRngCore + SeedableRng + CryptoRng,
       Rsdr: RngCore + CryptoRng {}
 
 
-#[cfg(all(feature="std", unix, not(target_os="emscripten")))]
+#[cfg(all(unix, not(target_os="emscripten")))]
 mod fork {
     extern crate libc;
 
-    use std::sync::atomic::{AtomicUsize, ATOMIC_USIZE_INIT, Ordering};
-    use std::sync::atomic::{AtomicBool, ATOMIC_BOOL_INIT};
+    use core::sync::atomic::{AtomicUsize, ATOMIC_USIZE_INIT, Ordering};
+    use core::sync::atomic::{AtomicBool, ATOMIC_BOOL_INIT};
 
     // Fork protection
     //
@@ -323,7 +323,7 @@ mod fork {
     }
 }
 
-#[cfg(not(all(feature="std", unix, not(target_os="emscripten"))))]
+#[cfg(not(all(unix, not(target_os="emscripten"))))]
 mod fork {
     pub fn get_fork_counter() -> usize { 0 }
     pub fn register_fork_handler() {}

